.github/workflows/lighty-rnc-app/test-lighty-rnc-app-cluster.sh

```diff
@@ -0,0 +1,219 @@
+CTRL0_NAME=${KUB_NAMES[0]}
+CTRL1_NAME=${KUB_NAMES[1]}
+CTRL2_NAME=${KUB_NAMES[2]}
+
+CONTROLLER_PORT=8888
+SIMULATOR_PORT=17830
+HTTP_STATUS_CODES=("200" "201" "202" "204")
+declare -a test_results
+
+echo "Controller0 IPS set to: " $CTRL0_IP " running in pod name: " $CTRL0_NAME
+echo "Controller1 IPS set to: " $CTRL1_IP " running in pod name: " $CTRL1_NAME
+echo "Controller2 IPS set to: " $CTRL2_IP " running in pod name: " $CTRL2_NAME
+
+# Start lighty-netconf-simulator in minikube network
+docker build -t lighty-netconf-simulator ${GITHUB_WORKSPACE}/.github/workflows/lighty-rnc-app/simulator
+docker run -d --rm --name netconf-simulator -p$SIMULATOR_PORT:$SIMULATOR_PORT lighty-netconf-simulator:latest
+
+# Get netconf-simulator container's IP
+SIMULATOR_IP=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' netconf-simulator)
+
+# List pods
+minikube kubectl -- get pods
+
+# List Services
+minikube kubectl -- get services
+
+# check if simulator has opened port
+for i in {1..20} ; do
+    nc -z $SIMULATOR_IP $SIMULATOR_PORT
+    if [ $? ]
+    then
+      break;
+    fi
+    echo "counter: $i"
+    sleep 1
+done
+
+printLine() {
+  printf '%.0s-' {1..100}; echo ""
+}
+
+assertHttpStatusCode() {
+  printLine
+  if [[ ${HTTP_STATUS_CODES[*]} =~ "$1" ]]
+  then
+    echo -e "HTTP request methods: $2\nURL: $3\nStatus Code: $1\nTest passed\n"
+    test_results+=(0)
+  else
+    echo -e "HTTP request methods: $2\nURL: $3\nStatus Code: $1\nTest failed\n"
+    test_results+=(1)
+  fi
+}
+
+assertNodeConnected() {
+  if [[ $1 =~ .*"connected".* ]]
+  then
+    echo "connected"
+  else
+    echo "$1"
+  fi
+}
+
+assertPodsTopologyResponse() {
+  local previousResponse=""
+  for pod_controller_ip in $POD_CONTROLLER_IPS;
+  do
+    TOPOLOGY_RESPONSE=$(curl --request GET \
+      'http://'$pod_controller_ip:$CONTROLLER_PORT'/restconf/data/network-topology:network-topology')
+      if [[ -z "$TOPOLOGY_RESPONSE" ]]
+      then
+        echo "Empty response from pod ip: " $pod_controller_ip
+        test_results+=(1)
+      elif [[ -z "$previousResponse" ]]
+      then
+        # First request
+        echo "first response" $TOPOLOGY_RESPONSE
+        previousResponse=$TOPOLOGY_RESPONSE
+      elif [[ "$previousResponse" != "$TOPOLOGY_RESPONSE" ]]
+      then
+        echo "Previous response doesn't match: [ $previousResponse ] wtih current resposne : [ $TOPOLOGY_RESPONSE ] "
+        test_results+=(1)
+      fi
+      echo "Succes compare for IP " $pod_controller_ip
+  done
+}
+
+printLine
+echo "-- Lighty-rcgnmi-app curl tests --"
+
+# Cluster state (:8558/cluster/members)
+for pod_controller_ip in $POD_CONTROLLER_IPS; \
+do \
+  assertHttpStatusCode $(curl -o /dev/null -s -w "%{http_code} GET %{url_effective}\n" \
+   -H "Content-Type: application/json" \
+   http://$pod_controller_ip:8558/cluster/members) \
+;done
+sleep 1
+
+# Pods healthcheck (:8888/restconf/operations)
```

Comment:
We already have liveness readiness probe. Do we really need this?

Reply:
Removed